global_ids.md

```diff
@@ -0,0 +1,187 @@
+# Global Identifiers
+
+## Goal
+
+Simplify Pachyderm's data model, both from a user and implementation
+perspective by having global identifiers for jobs and commits, rather than
+local identifiers that are linked together.
+
+## Use Case:
+
+Give users intuitive answers to the question how do I get from `X` to `Y`,
+     where `X` and `Y` can be commits or jobs at any point in the DAG.
+
+## Proposal
+
+When you commit data to Pachyderm your new commit has an ID associated with it.
+This ID is the closest thing to a global identifier for what users think of as
+the DAG / job associated with that data. Using this identifier you can do a
```

Comment:
In talking about this idea with @lukemarsden, it's occurred to me that conceptualizing global IDs as DAG snapshots makes it a lot easier to see how a data scientist might do experiments: fork the dag, make changes (create pipelines/commit data) and then merge with the "master" DAG.

A DAG snapshot could include the head commit of every repo/pipeline. Merge conflicts could look like and be handled similarly to merge conflicts in Git (with repos analogous to files in Git): if `mybranch` and `master` both committed to the same repo or both updated the same pipeline, the user creating the merge commit must choose, for each conflict, the head commit for that repo/pipeline from either `mybranch` or `master`.

Reply:
(as you point out below, part of the benefit of this design is that it reduces our etcd usage—anchoring all of changes to a single file in the spec repo wouldn't accomplish that goal, so it may not make sense as an implementation, but I've found it conceptually helpful)